app/helpers/cms_helper.rb

```diff
@@ -154,45 +154,41 @@ def cta_live_report
   end
 
   def get_resource_links 
-    resources = [
+    [
       get_resource(:resource_link_text, :resource_link_url, 'link', 'link-external'),
       get_resource(:resource_file_title, :resource_file, 'download', 'download')
     ].compact
   end
 
-  # Turns link[:url] value into a valid link if no http:// or https:// supplied
-  # Also sanitises it in the case of a download link
-  def linkify(url, fragment_link)
-    # If it is a file as opposed to a link
-    if fragment_link =~ /(file)/ 
-      file = @cms_page.fragments.find_by(tag: 'file')
-      return '' if !file || file.attachments.first.blank?
-      if Rails.env.development?
-        return rails_blob_path(file.attachments.first)
-      else
-        return file.attachments.first.service_url&.split('?')&.first
-      end
-    end
-    # TODO - add validations to Comfy pages controller to make it more robust 
-    # This is only temporary 
-    url =~ /(http:|https:)/ ? url : 'https://' + url
-  end
-
   private
 
   #  variation of resource link, if it is just a link, or whether it is a file
   # if those specific fragments for that page are present
   def get_resource(fragment_text, fragment_link, button_text, button_class)
-    if cms_fragment_content(fragment_text, @cms_page) && cms_fragment_content(fragment_link, @cms_page)
-      {
-        button: I18n.t("global.button.#{button_text}"),
-        classes: "button--#{button_class}",
-        text: cms_fragment_render(fragment_text, @cms_page),
-        title: cms_fragment_render(fragment_text, @cms_page),
-        url: linkify(cms_fragment_render(fragment_link, @cms_page), fragment_link)
-      }
+    frag_text, frag_link = @cms_page.fragments.where(identifier: [fragment_text, fragment_link])
+    return unless frag_text || frag_link
+
+    {
+      button: I18n.t("global.button.#{button_text}"),
+      classes: "button--#{button_class}",
+      text: cms_fragment_render(fragment_text, @cms_page),
+      title: cms_fragment_render(fragment_text, @cms_page),
+      url: linkify(frag_link)
+    }
+  end
+
+  # Turns link[:url] value into a valid link if no http:// or https:// supplied
+  # Also sanitises it in the case of a download link
+  def linkify(fragment_link)
+    if fragment_link.tag == 'file'
+      if Rails.env.development?
+        return rails_blob_path(fragment_link.attachments.first)
+      else
+        return fragment_link.attachments.first.service_url&.split('?')&.first
+      end
     else
-      nil
+      url = fragment_link.content
+      url =~ /(http:|https:)/ ? url : 'https://' + url
```

Comment:
Should we continue to have this line in? This was only a temporary fix to get round the issue of valid URLs, I feel that if you are going to make a change to the methods, then this would also be a good opportunity to make this a more robust fix 😉

Reply:
Thanks, I've updated this to double check the URL string and return an 'Invalid' string if not valid.
Could you and/or @stacytalbot double check that the partial related to this helper is actually being used and where? I found it difficult to test this properly as I get confused between the `partials/cards/_resources.html.erb` partial (where links and files are showing properly although using different CMS tags to fetch those) and the `layouts/cms/partia/_resource.html.erb` where differently named tags are meant to be used.

Reply:
That is the `layouts/cms/partial/_resource.html.erb` which concerns individual resource pages, and is used for them - not the `partials/cards/_resources.html.erb` which you can think of as the index view of all the resources present in the CMS.

Reply:
Thanks for claryfing!
So the question (perhaps more for @stacytalbot ) is whether files and links in the individual resource pages should be using different CMS tags (perhaps referring to different things) than the tags used for the home page?

Reply:
Perfectly clear now, thanks!